packages/app/src/cli/models/app/app.ts

```diff
@@ -252,6 +271,32 @@ export class App implements AppInterface {
   }
 }
 
+export function validateFunctionExtensionsWithUiHandle(
+  functionExtensionsWithUiHandle: ExtensionInstance<FunctionConfigType>[],
+  allExtension: ExtensionInstance[],
+): string[] | undefined {
+  const errors: string[] = []
+
+  functionExtensionsWithUiHandle.forEach((extension) => {
+    const uiHandle = extension.configuration.ui?.handle!
+
+    const matchingExtension = findExtensionByHandle(allExtension, uiHandle)
+    if (!matchingExtension) {
+      errors.push(`[${extension.name}] - Local app must contain a ui_extension with handle '${uiHandle}'`)
+    } else if (matchingExtension?.configuration?.type !== 'ui_extension') {
```

Comment:
Nit: `matchingExtension` should not be falsey here

```suggestion
    } else if (matchingExtension.configuration?.type !== 'ui_extension') {
```